napalm_junos/junos.py

```diff
@@ -854,20 +854,38 @@ def get_interfaces_ip(self):
 
     def get_mac_address_table(self):
 
-        mac_address_table = list()
+        mac_address_table = []
 
         mac_table = junos_views.junos_mac_address_table(self.device)
+
+        if self.device.facts.get('personality', '') in ['SWITCH']:  # for EX & QFX devices
```

Comment:
Hum.. maybe, but would be loaded even when the device is not EX/QFX. But would be cleaner indeed.